### ISSUE TITLE
[Crash]: crash while investigating issue #9736: looking at weird behaviour on the mutliplayer server list menu.

### ISSUE THREAD

**Issue by Azusa257** [OpenTTD/OpenTTD#9737]
### Version of OpenTTD

Version: 12.1, OS: Windows 10

### Steps to reproduce

Game crashed on while investigating issue #9736.
Steps to reproduce: unknown.
I reproduced the above issue and left the game in that state, and when I returned, the game had crashed after few minutes of inactivity. 

### Upload crash files

[OpenTTD_crash_20211206.zip](https://github.com/OpenTTD/OpenTTD/files/7661536/OpenTTD_crash_20211206.zip)

**Comment by glx22** [comment]
```
>	openttd.exe!ClientNetworkGameSocketHandler::ClientError(NetworkRecvStatus) Ligne 213	C++
 	openttd.exe!ClientNetworkGameSocketHandler::Receive() Ligne 243	C++
 	openttd.exe!NetworkGameLoop() Ligne 1038	C++
 	openttd.exe!GameLoop() Ligne 1461	C++
 	openttd.exe!VideoDriver::GameLoop() Ligne 37	C++
 	openttd.exe!VideoDriver::GameThread() Ligne 46	C++
 	openttd.exe!StartNewThread::__l3::<lambda>(const char * name, void(*)(VideoDriver *) && F, VideoDriver * && <A_0>) Ligne 65	C++
 	[Code externe]	
 	openttd.exe!thread_start<unsigned int (__cdecl*)(void *),1>(void * const parameter) Ligne 97	C++
 	[Code externe]	
```
`this` is `nullptr`

**Comment by TrueBrain** [comment]
With 14.0 we added a "deferred deletion" for the GameSocket entries, which should also address this issue. At least, looking at where it crashed and how it could get there, it totally makes sense that it could happen. And that scenario is fixed in 14.0.

So going to close up this issue!
